### PR TITLE
fix(source-google-ads): fix attribute error on check connection

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
resolved: https://github.com/airbytehq/oncall/issues/6141

## How
We should handle AirbyteTracedException on check separate from Google Ads Exception to avoid attribute error when AirbyteTracedException is passed to traced_exception method.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py [hunk 1]
```diff
@@ -225,6 +225,8 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                         customer_id=customer.id,
                         login_customer_id=customer.login_customer_id,
                     )
+                except AirbyteTracedException as e:
+                    raise e
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:229]: I'm a little bit worried about losing the underlying context here, probably we should raise something like:

```python
raise e from e
```

But if it is OK, and we don't lose the context of the `AirbyteTracedException` raised prior, we should be fine with this.

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py (hunk 1 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py]: Thanks, updated

**Review by maxi297 — APPROVED** [review]
LGTM!

**Pull request merged**